### PR TITLE
Add the Valgrind memory check for examples

### LINKED ISSUE #67: Integrate valgrind memory check into github/actions
We'd better integrate valgrind memory check into actions in this repo.

e.g.

```shell
valgrind --tool=memcheck --leak-check=full ./create_pod_bin
```

### PR DESCRIPTION
**Pull request by ityuhui**
Close #67

### CODE REVIEW

File: examples/Makefile [hunk 1]
```diff
@@ -35,4 +35,15 @@ test:
 	cd generic; make test
 	cd multi_thread; make test; sleep 10
 	kubectl describe po test-pod-8
-	cd exec_pod; make test
\ No newline at end of file
+	cd exec_pod; make test
+
+memcheck:
+	cd create_pod; make memcheck; sleep 10
```
Comment (brendandburns) [inline examples/Makefile:41]: Why are there sleeps here (and  below)?

Comment (ityuhui) [reply examples/Makefile:41]: I want to give the Kubernetes components (e.g. kubelet, API server) some time to pull the image and initialize the pod when creating a pod. Prevent possible errors in the following examples (e.g. deleting the pod, executing command in the pod when the pod is not initialized).

Comment (brendandburns) [reply examples/Makefile:41]: Perhaps move the `sleep 10` to it's own line and add a comment?

```
memcheck:
      sleep 10 # wait for kubernetes components to come up
     cd create_pod; make memcheck
      ...

```

wdyt?

Comment (ityuhui) [reply examples/Makefile:41]: It's a good idea. And I find a better "waiting" solution:

```bash
kubectl wait --for=condition=ready --all pod -n namespace --timeout=30s
```

Let me have a try today.

Comment (ityuhui) [reply examples/Makefile]: Done. Please take another look.

**Pull request merged**